src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1730,6 +1730,22 @@ bool checkValid(MethodSymbol candidate, bool isCandidate, DiagnosticBag specific
                         }
                     }
                 }
+                else if (LanguageVersion >= MessageID.IDS_FeatureAsyncMain.RequiredVersion() && taskEntryPoints.Count > 0 && viableEntryPoints.Count == 1)
```

Comment:
It feels like it is valid to warn when `validEntryPoints.Count > 0`. Otherwise the user may address the ambiguity among the sync Main methods in their project, rebuild, and then potentially discover new warnings about async Main methods in their program. Might as well just give them all the diagnostics in the first run.

Reply:
This is a good point: it's not really suppressing a cascaded diagnostic, it's the same issue but a different diagnostic because compat.

---
In reply to: [473358297](https://github.com/dotnet/roslyn/pull/46832#discussion_r473358297) [](ancestors = 473358297)